lib/Sitemap/Document/DocumentTreeGenerator.php

```diff
@@ -145,7 +145,7 @@ private function visit(Document $document, DocumentGeneratorContext $context): \
             $document = Document\Hardlink\Service::wrap($document);
```

Comment:
since it is specific to hard links, it is better to check for document is empty here and return.